apps/web/src/app/(unauthenticated)/unverified-account/page.tsx

```diff
@@ -0,0 +1,68 @@
+'use client';
+
+import { useState } from 'react';
+
+import { useSearchParams } from 'next/navigation';
+
+import { Mails } from 'lucide-react';
+
+import { ONE_SECOND } from '@documenso/lib/constants/time';
+import { trpc } from '@documenso/trpc/react';
+import { Button } from '@documenso/ui/primitives/button';
+import { useToast } from '@documenso/ui/primitives/use-toast';
+
+const RESEND_CONFIRMATION_EMAIL_TIMEOUT = 20 * ONE_SECOND;
+
+export default function UnverifiedAccount() {
+  const [isButtonDisabled, setIsButtonDisabled] = useState(false);
+  const searchParams = useSearchParams();
+  const { toast } = useToast();
+
+  const encryptedEmail = searchParams?.get('token') ?? '';
+
+  const { mutateAsync: sendConfirmationEmail } = trpc.profile.sendConfirmationEmail.useMutation();
+
+  const onResendConfirmationEmail = async () => {
+    try {
+      setIsButtonDisabled(true);
+
+      await sendConfirmationEmail({ encryptedEmail });
+
+      toast({
+        title: 'Success',
+        description: 'Verification email sent successfully.',
+        duration: 5000,
+      });
+
+      setTimeout(() => setIsButtonDisabled(false), RESEND_CONFIRMATION_EMAIL_TIMEOUT);
```

Comment:
I think it would be more intuitive to the user if we do something like this instead

- Store a time of when the last email was sent
- Allow the button to always be clickable unless we are in the process of sending an email

Then at the start of this function, if the cooldown period is still active we can toast saying they need to wait X amount

-------------

Alternatively, we can display something in the UI saying please wait X amount of time

I'm unsure which is better, or if there are other solutions but imo there should be some sort of indicator to the user